From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Matti Viljanen <matti.viljanen@jolla.com>
Date: Sun, 12 Oct 2025 18:16:32 +0300
Subject: [PATCH] Fix host triplet generation

CMake makes the host triplet "{i486|armv7l|aarch64}-meego-linux"
whereas config.guess provided "i686-pc-linux-gnu",
"armv7l-unknown-linux-gnueabi" and "aarch64-unknown-linux-gnu".
The rest of our build system can't handle the changes,
so we have to replicate the GNU config.guess behavior.
---
 CMakeLists.txt | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7808115c1..bfaedf8a7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -143,12 +143,27 @@ function(makemacros)
 	set(host_cpu ${CMAKE_HOST_SYSTEM_PROCESSOR})
 	string(TOLOWER ${CMAKE_HOST_SYSTEM_NAME} host_os)
 	set(host_vendor ${RPM_VENDOR})
-	set(host ${host_cpu}-${host_vendor}-${host_os})
 
 	set(RPMCANONVENDOR ${host_vendor})
 	set(RPMCANONOS ${host_os})
 	set(RPMCANONGNU -gnu)
 
+	if (host_cpu STREQUAL "i486" OR host_cpu STREQUAL "i686")
+		set(host_cpu "i686")
+		set(host_vendor "pc")
+	else()
+		set(host_vendor "unknown")
+	endif()
+
+	execute_process (
+		COMMAND bash -c "if [ `echo | gcc -dM -E - | grep -c __ARM_EABI__` -gt 0 ]; then echo eabihf; fi"
+		OUTPUT_STRIP_TRAILING_WHITESPACE
+		OUTPUT_VARIABLE ABI
+	)
+	string(CONCAT host_abi "gnu" ${ABI})
+
+	set(host ${host_cpu}-${host_vendor}-${host_os}-${host_abi})
+
 	if (ENABLE_CUTF8)
 		set(C_LOCALE "C.UTF-8")
 	else()
